spring-integration-core/src/main/java/org/springframework/integration/transformer/EncodingPayloadTransformer.java

```diff
@@ -36,7 +36,7 @@ public EncodingPayloadTransformer(Codec codec) {
 	}
 
 	@Override
-	protected byte[] transformPayload(T payload) throws Exception {
+	protected byte[] transformPayload(T payload) throws Exception {//NOSONAR
```

Comment:
@garyrussell, to be honest I don't like this restriction.
Seems for me we lived with such a design for a long time already.
So, how about to fix SONAR rules a bit on the matter?